File: Algorithms/Graph/Breadth First Search/Breadth First Search.cpp
typedef pair<int, int> ii;

void BFS(vector<ii> adj[], int V)
{

    bool vis[V];
    fill(vis, vis + V, false);
    queue<int> q;
    q.push(0);

    while (!q.empty())
    {
        int u = q.front();
        q.pop();

        if (vis[u])
            continue;
        vis[u] = true;

        for (ii edge : adj[u])
        {
            if (!vis[edge.first])
                q.push(edge.first);
        }
    }
}

File: Algorithms/Graph/Breadth First Search/Breadth First Search.py
from collections import deque


def BFS(adj):
    V = len(adj)

    vis = [False] * V
    q = deque()
    q.append(0)
    while q:
        u = q.popleft()

        if vis[u]:
            continue
        vis[u] = True

        for v, w in adj[u]:
            if not vis[v]:
                q.append(v)
